sdk/servicebus/azure-servicebus/azure/servicebus/_common/receiver_mixins.py

```diff
@@ -22,14 +24,21 @@
 
 
 class ReceiverMixin(object):  # pylint: disable=too-many-instance-attributes
-    def _create_attribute(self, **kwargs):
+    def _populate_attributes(self, **kwargs):
         if kwargs.get("subscription_name"):
             self._subscription_name = kwargs.get("subscription_name")
             self._is_subscription = True
             self.entity_path = self._entity_name + "/Subscriptions/" + self._subscription_name
         else:
             self.entity_path = self._entity_name
 
+        if kwargs.get("is_dead_letter_receiver", False):
```

Comment:
So I did not communicate myself well when mentioning the change I made here to you:

The part that I liked from my PR was bringing these bits 
```
        self._message_iter = None
        self._connection = kwargs.get("connection")
        self._prefetch = kwargs.get("prefetch")
```
into the populate stage.

I was actually less confident with bringing the entity_path manipulation inside/kinda liked how you did it outside, since "is dead letter" is a little bit external to a "generic receiver" (as opposed to it blindly connecting to whatever entity path we give, having done the mutation prior.)  Having this mutation local to the get_queue_deadletter_receiver function, although it creates a tiny bit of overhead (and this is why I suggested helperifying this construction) makes more clear what the ACTUAL difference is between a deadletter receiver and not without the user having to dig in to this layer as well.  I'll leave it up to you which location you prefer, we're not tied to this decision (totally internal) and as usual it's very bikesheddy, but figured I'd share my thoughts.

Reply:
I'm afraid we still need to declare the instance variables  in `__init__` otherwise we would get pylint error:

"Attribute '_message_iter' defined outside __init__"

Reply:
So for the discussion on "inside" VS "outside" constructing a dead letter path:

- If user is using `ServiceBusClient` to create DLQ receiver, then "inside" VS "outside" is just a matter of internal implementation.

- If user is creating DLQ receiver directly by creating `ServiceBusReceiver`, then "inside" VS "outside" matters.
    - "outside" means we need to provide path generation helper method, user would use the helper method to generate a DLQ path first, then manually pass the path into the receiver.
    - ”insde" means we provide options on the constructor and do the work for users.

So my concern is the "outside" approach definitely brings clarity, but is the clarity that users do care about (want to learn about) -- do they really want to know the detail of how a (Transfer)DLQ path is composed or they just want an option which allows them to connect to DQL queue.

Reply:
as per OOB discussion, moved the DLQ path generation outside.

Suppressed the pylint warning `Attribute '_message_iter' defined outside __init__`, `populate_attributes` is responsible for initializing receiver instance variables for clarity.